Sankey docs (#2090)
telerik/blazor-docs: components/sankeydiagram/customization/labels.md

---
title: Labels
page_title: Sankey Diagram Labels
description: Sankey Diagram Labels
slug: sankey-diagram-labels
tags: telerik,blazor,sankey,diagram,chart,labels
published: True
position: 5
---

# Sankey Diagram Labels

> The Sankey Diagram Lables are the names of the [nodes]{%slug sankey-diagram-nodes%} that can be displayed on the nodes or next to them. This article explains how to customize the labels in the UI for Blazor Sankey Diagram. The listed settings will be applied to all labels in the Sankey Diagram.

Reviewer: the names of the **nodes** that can be displayed on the **nodes** or next to them - this is hard to understand and at first glance looks like an endless recursion. Consider rewording, for example.

```suggestion
The Sankey Diagram Labels represent the [node]{%slug sankey-diagram-nodes%} names. The labels can show over the nodes or next to them. This article explains how to customize the labels in the UI for Blazor Sankey Diagram. Each setting applies to all labels in the Sankey Diagram.
```

## Basic Customization

> To customize the labels, declare the `<SankeyLabels>` tag as a direct child of `<TelerikSankey>`. It exposes the following parameters:

Reviewer: "It" is ambiguous.
```suggestion
To customize the labels, declare a `<SankeyLabels>` tag as a direct child of `<TelerikSankey>`. The `<SankeyLabels>` tag has the following parameters:
```

| Parameter | Type | Description |
| --------- | ---- | ----------- |
| `Left` | `double?` | The left offset of the labels. |
| `Top` | `double?` | The top offset of the labels. |

### Padding

The `<SankeyLabelsPadding>` child tag provides the following properties:

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type | Description |
| --------- | ---- | ----------- |
| `Top` | `double?` | The top padding of the labels. |
| `Bottom` | `double?` | The bottom padding of the labels. |
| `Left` | `double?` | The left padding of the labels. |
| `Right` | `double?` | The right padding of the labels. |

### Stroke

The `<SankeyLabelsStroke>` child tag provides the following properties:

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type | Description |
| --------- | ---- | ----------- |
| `Color` | `string` | The color of the stroke. |
| `LineJoin` | `string` | The [line join](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-linejoin) of the stroke. |
| `Left` | `double?` | The width of the stroke. |

## Example

> Here is an example customization of the labels in the Sankey Diagram.

Reviewer: Convert to a `>caption`

Author: This is deliberately not a caption as it is the only text in this section (similar to https://docs.telerik.com/blazor-ui/components/popover/position-collision#example)

@code {
    private SankeyData Data { get; set; }
    private string EventLog { get; set; } = string.Empty;

    #region Data generation

    protected override void OnInitialized()
    {
        Data = new SankeyData()
            {
                Nodes = new SankeyDataNodes(),
                Links = new SankeyDataLinks()
            };

        Data.Nodes.Add(new SankeyDataNode() { Id = 1, Label = new SankeyDataNodeLabel() { Text = "Tablet (12%)" } });
        Data.Nodes.Add(new SankeyDataNode() { Id = 2, Label = new SankeyDataNodeLabel() { Text = "Mobile (40%)" } });
        Data.Nodes.Add(new SankeyDataNode() { Id = 3, Label = new SankeyDataNodeLabel() { Text = "Desktop (48%)" } });
        Data.Nodes.Add(new SankeyDataNode() { Id = 4, Label = new SankeyDataNodeLabel() { Text = "< 18 years (8%)" } });
        Data.Nodes.Add(new SankeyDataNode() { Id = 5, Label = new SankeyDataNodeLabel() { Text = "18-26 years (35%)" } });
        Data.Nodes.Add(new SankeyDataNode() { Id = 6, Label = new SankeyDataNodeLabel() { Text = "27-40 years (38%)" } });
        Data.Nodes.Add(new SankeyDataNode() { Id = 7, Label = new SankeyDataNodeLabel() { Text = "> 40 years (19%)" } });

>         Data.Links.Add(new SankeyDataLink() { SourceId = 1, TargetId = 4, Value = 4 });

Reviewer: Not urgent for this release, but consider generating random data in a loop.
